engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
Wrote a sample application and tested this change by killing the containers. There was no issue and the recovery started from the expected window id ( committed + 1 ). Still not sure why we have to keep the window data of the committed window.